scipy/stats/_qmc.py

```diff
@@ -0,0 +1,1500 @@
+    iterative : bool, optional
+        Must be False if not using it for updating the discrepancy.
+        Default is False. Refer to the notes for more details.
+    method : str, optional
+        Type of discrepancy, can be ['CD', 'WD', 'MD', 'star']. Refer to
+        the notes for more details. Default is ``CD``.
+
+    Returns
+    -------
+    discrepancy : float
+        Discrepancy.
+
+    Notes
+    -----
+    The discrepancy is a uniformity criterion used to assess the space filling
+    of a number of samples in a hypercube.
+    The discrepancy measures how the spread of the points deviates from a
+    uniform distribution.
+    The lower the value is, the better the coverage of the parameter space is.
+
+    A discrepancy quantifies the distance between the continuous uniform
+    distribution on a hypercube and the discrete uniform distribution on
+    :math:`n` distinct sample points. Smaller values are better. For a
+    collection of subsets of the hypercube, the discrepancy is the greatest
+    absolute difference between the fraction of sample points in one of those
+    subsets and the volume of that subset. There are different definitions of
+    discrepancy corresponding to different collections of subsets. Some
+    versions take a root mean square difference over subsets instead of
+    a maximum.
+
+    A measure of uniformity is reasonable if it satisfies the following
+    criteria [1]_:
+
+    1. It is invariant under permuting factors and/or runs.
+    2. It is invariant under rotation of the coordinates.
+    3. It can measure not only uniformity of the sample over the hypercube,
+       but also the projection uniformity of the sample over non-empty
+       subset of lower dimension hypercubes.
+    4. There is some reasonable geometric meaning.
+    5. It is easy to compute.
+    6. It satisfies the Koksma-Hlawka-like inequality.
+    7. It is consistent with other criteria in experimental design.
+
+    Four methods are available:
+
+    * ``CD``: Centered Discrepancy - subspace involves a corner of the
+      hypercube
+    * ``WD``: Wrap-around Discrepancy - subspace can wrap around bounds
+    * ``MD``: Mixture Discrepancy - mix between CD/WD covering more criteria
+    * ``star``: Star L2-discrepancy - like CD BUT variant to rotation
+
+    Lastly, using ``iterative=True``, it is possible to compute the
+    discrepancy as if we had :math:`n+1` samples. This is useful if we want
+    to add a point to a sampling and check the candidate which would give the
+    lowest discrepancy. Then you could just update the discrepancy with
+    each candidate using `_update_discrepancy`. This method is faster than
+    computing the discrepancy for a large number of candidates.
+
+    References
+    ----------
+    .. [1] Fang et al. Design and modeling for computer experiments,
+       Computer Science and Data Analysis Series, 2006.
+    .. [2] Zhou Y.-D. et al. Mixture discrepancy for quasi-random point sets
```

Comment:
```suggestion
    .. [2] Zhou Y.-D. et al. Mixture discrepancy for quasi-random point sets.
```
Nitpick, I know, so it's not a blocker, but I have to mention it:

Although we're not super picky about citation format in the documentation, most major citation formats use periods instead of commas between article and journal name.

![image](https://user-images.githubusercontent.com/6570539/104663662-54994480-5682-11eb-976f-e5d3fe045412.png)

And it would be nice to have something to visually distinguish between article name (e.g. in quotes) and journal name (e.g. in italics) here and throughout references in the submodule.

Reply:
What you changed it to is fine. I'll probably just put quotes around the title of "Mixture discrepancy..." as you did for the other two, to make it consistent.